Mathlib/NumberTheory/NumberField/Units/DirichletTheorem.lean

```diff
@@ -478,6 +482,11 @@ theorem fundSystem_mk (i : Fin (rank K)) :
     Quotient.out', Quotient.out_equiv_out]
   rfl
 
+theorem logEmbedding_fundSystem (i : Fin (rank K)) :
+    logEmbedding K (fundSystem K i) = basisUnitLattice K i := by
+   rw [basisUnitLattice, Basis.map_apply, ← fundSystem_mk, ← logEmbeddingEquiv_apply]
+   rfl
```

Comment:
This and the previous one end with `rfl`. Are we missing some lemmas here?

Reply:
Yes, it is a bit of a mystery to me. I think we're missing something like `Additive.toMul x = Additive.toMul y → x = y` and similar results (whose proofs is essentially `rfl` too) but I am not sure they're worth adding. The mess in `fundSystem_mk` comes also from the fact that there are two kinds of `Additive.toMul` involved...

Reply:
I have the impression you are abusing `defeq`: `logEmbedding` takes a term of type `Additive (𝓞 K)ˣ`, but you are giving it `fundSystem K i`, which has type `(𝓞 K)ˣ`. Can we fix this somehow?

Reply:
Yeah, you're right. I am probably a bit sloppy with my use of `Additive`. I'll see if I can fix that

Reply:
All right, I think I took of all the abuses of defeq. There was a lot of `Additive.ofMul` missing but also the use `⟦x⟧` rather than `QuotientGroup.mk x` to go from `(𝓞 K)ˣ` to `((𝓞 K)ˣ ⧸ (torsion K))` that resulted in a wrongly typed quotient.